### DISCUSSION TITLE
Slow Modals & Slow in General

### DISCUSSION THREAD

**Discussion by zuroma999** [Budibase/budibase#9482] [Help] [answered]
I know there are a few outstanding issues relating to Budibase speed, but I'm wondering if they're related or not to  the slowdowns to what I'm seeing.

I created a free-tier account on Budibase's cloud hosting. Not sure which version of Budibase gets installed. Trying the "Business Directory Template", without modification. It takes about 5-7 seconds to load a modal when clicking to view a row.

With my test app ([https://gforg.budibase.app/app/test-app](https://gforg.budibase.app/app/test-app)) I have a page with just 2 repeaters, and small test tables, so in total it should fetch 2-3 rows, yet the modal is still taking up to 10s to finish rendering.

We're very excited about Budibase, but just curious if this is how fast it should be running?

**Comment by rolandwang19** [comment]
I'm experiencing the same. Just loading a simple form and taking a long time as well.

**Comment by gmentzik** [comment]
I have a component in a page that is a provider with repeater fetching and id and then inside its repeater element another provider with repeater filtering with that id that has an update form inside each repeater element. Loading of forms takes a long time. More specifically there is **high cpu usage on local machine** for more than 10seconds and then page starts to fetch data and load. At a page where I have multiple checkboxes in repeaters I even get an unresponsive warning from browser because of the delay.
I did not have any issue at all with the Budibase version 2.1.42 before new loading effect. Yesterday I updated to version 2.3.16 and all pages with update forms became slow to load. Especially those that forms are inside a repeater. For now reverted back to 2.1.42. Page loading is really fast at that version. Please consider reverting to old loading page design! New page loading with placeholders has caused many speed issues!

**Comment by gmentzik** [comment, marked as answer]
@melohagan Hi sorry for referring to you, but I believe no attention is given at that issue. Are any actions taken?

**Reply by melohagan** [reply]
Hey @gmentzik 

In the next release (March 6th) the skeleton loader is going to be rolled back, which should hopefully resolve the performance issues.

**Reply by gmentzik** [reply]
That's a relief, because i needed to redesign a few pages to make update work. Thank you! Waiting for next release then!

**Reply by gmentzik** [reply]
Hi @melohagan,
I just updated my app to 2.4.8 and skeleton loader is still there still causing issues to me. ANy updates on that?

**Reply by melohagan** [reply]
Hey @gmentzik I got my timing wrong, sorry about that. It will definitely be removed in the next release. It's being worked on right now.

**Reply by gmentzik** [reply]
Hi @melohagan I just updated budibase and app to 2.4.43 and still see skeleton loader. Any news?

**Reply by melohagan** [reply]
Hey @gmentzik The next with the skeleton loader removed will be **2.5.X** which will be released this week; hopefully today at some stage!

**Reply by gmentzik** [reply]
Hi @melohagan. I have updated Budibase  and my app to 2.5.0. Skeleton loader is gone and app seems to be working properly. At least I see no issues that I had with skeleton loader so far. Thank you all for your hard effort!

**Reply by melohagan** [reply]
That's great to hear - thank you for the nice words 😄
